### DISCUSSION TITLE
How to run self-hosted renovate with gitlab?

### DISCUSSION THREAD

**Discussion by sananibrahimov-finocomp** [renovatebot/renovate#9292] [Request Help] [answered]
Trying to run self hosted renovate within a docker container.

```
config.js
module.exports = {
                    platform: 'gitlab',
                    endpoint: 'https://gitlab.com/api/v4/',
                    token: 'xxxxx',
                    gitAuthor: 'Renovate Bot <myemail.co.uk>',
                    labels: ['renovate', 'dependencies', 'automated'],
                    assignees: ['myuser'],
                    onboarding: true,
                    enabledManagers: [
                                                        'gradle',
                                                        'npm'
                                                    ],
                    onboardingConfig: {
                                              extends: ['config:base'],
                                            },
                    repositories: [
                                                'myuser/application-template'
                                              ],
};

```
Command to run 
`$ docker run --rm -v "config.js:/usr/src/app/config.js" renovate/renovate`
FATAL: You must configure a GitHub personal access token
 INFO: Renovate is exiting with a non-zero code due to the following logged errors
       "loggerErrors": [
         {
           "name": "renovate",
           "level": 60,
           "logContext": "SzVYmGYzy",
           "msg": "You must configure a GitHub personal access token"
         }
       ]

Why do I need a gitgub token? I have nothing to do with github I have already provided gitlab token.
What am I doing wrong?

**Comment by sananibrahimov-finocomp** [comment, marked as answer]
Resolved. Solution was - absolute path needed:
` docker run --rm -v ${PWD}/config.js:/usr/src/app/config.js renovate/renovate`

**Comment by viceice** [comment]
Please checkout our official gitlabci templates

https://gitlab.com/renovate-bot/renovate-runner
